File: src/ban.ts
import * as fs from 'fs';
import { tokenize } from './lib/tokenizer';
import { parse } from './lib/parser';
import { compile } from './lib/compiler';

const source = fs.readFileSync(process.argv[2], 'utf-8');

function baneval(source: string) {
  const tokens = tokenize(source);
  const program = parse(tokens);
  const compiledProgram = compile(program);
  eval(compiledProgram);
}

baneval(source);

File: ban.py
from constants import DEBUG
import sys
from tokenizer import (Tokenizer, print_tokens)
from parser import Parser
import sys
import machine
source_file = open(sys.argv[1], "r")
source = source_file.read() 
source_file.close()
tokens = Tokenizer(source).tokenize()
if DEBUG:
    print_tokens(tokens)
program = Parser(tokens).parse()
if DEBUG:
    for statement in program:
        print(statement)
machine.execute(machine.ExecutionContext(), program)
